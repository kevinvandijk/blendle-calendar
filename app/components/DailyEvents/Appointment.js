import React from 'react';

// h-calendar 2.0 compatible
const Appointment = (props) => {
  return (
    <dl className="DailyEvents-Appointment h-event" style={{ width: `${props.width}%`, height: `${props.height}%`, top: `${props.top}%`, left: `${props.left}%` }}>
      <dt className="DailyEvents-Appointment-time"><time className="dt-start" dateTime="2013-06-30 18:00"></time>-<time className="dt-end" dateTime=""></time></dt>
      <dd className="DailyEvents-Appointment-content">
        <h1 className="DailyEvents-Appointment-title p-name">{ props.title }</h1>
        <p className="DailyEvents-Appointment-description p-summary">Lorem ipsum dolor</p>
      </dd>
    </dl>
  );
};

export default Appointment;
