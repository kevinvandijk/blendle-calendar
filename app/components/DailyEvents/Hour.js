import React from 'react';
import { PropTypes } from '../../helpers';

const { hour, node } = PropTypes;

const Hour = (props) => {
  return (
    <li className="DailyEvents-Hour">
      <time>{ props.label }</time>
      { props.children }
    </li>
  );
};

Hour.propTypes = {
  label: hour.isRequired,
  children: node
};

export default Hour;
