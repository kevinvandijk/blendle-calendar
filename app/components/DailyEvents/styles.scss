.DailyEvents {
  width: 540px;
  height: 610px;
  background: #fff;

}

.DailyEvents-list {
  display: flex;
  flex-direction: column;
  overflow-y: hidden;
  padding: 20px 40px 22px;
  padding-top: 20px;
  height: 100%;
  list-style: none;

  &.DailyEvents-list--is-scrollable {
    overflow-y: auto;
  }
}

$paddingLeft: 4ch;
.DailyEvents-Hour {
  position: relative;
  padding-left: $paddingLeft;
  min-height: 55px;

  flex-grow: 1;

  &:after {
    position: absolute;
    top: 0;
    right: 0;
    left: $paddingLeft;
    height: 1px;
    background: rgba(#B5B5B5, .2);
    content: "";
  }


  > time {
    position: absolute;
    top: -.5em;
    left: 0;
    color: #B5B5B5;
    font-size: 12px;
    line-height: 1;
  }
}

.DailyEvents-Appointment {
  position: absolute;
  overflow: hidden;
  margin: 0;
  padding: 8px 20px 0;
  width: 100%;
  border-radius: 4px;
  background: #FF6255;

  &:after {
    position: absolute;
    right: 20px;
    bottom: 0;
    left: 20px;
    height: 10px;
    background: linear-gradient(to bottom, rgba(#FF6456, 0) 0%, rgba(#FF6456, 1) 100%);
    content: "";
  }
}

.DailyEvents-Appointment-content {
  margin: 0;
  padding: 0;
}

.DailyEvents-Appointment-title {
  overflow: hidden;
  margin: 0;
  padding: 0;
  color: #fff;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-weight: normal;
  font-size: 14px;
  line-height: 1.4;
}

.DailyEvents-Appointment-time {
  float: right;
  margin-left: 6px;
  color: rgba(#fff, .6);
  font-size: 14px;
  line-height: 1.4;
}

.DailyEvents-Appointment-description {
  margin: 0;
  padding: 0;
  color: rgba(#fff, .8);
  font-size: 12px;
  line-height: 1.5;
}
