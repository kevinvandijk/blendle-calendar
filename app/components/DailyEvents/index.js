import React from 'react';

import './styles.scss';

const formatHourFromStart = (startHour, i) => {
  // TODO
  return startHour + i;
};

const DailyEvents = (props) => {
  const { displayHours, startHour } = props;

  return (
    <div className="DailyEvents">
      <div className="DailyEvents-list">
        {[...Array(displayHours)].map((a, i) => (
          <li className="DailyEvents-hour">
            <time>{ formatHourFromStart(startHour, i) }</time>
          </li>
        ))}
      </div>

    </div>
  );
};

const isValidHour = (props, propName) => {
  // TODO
  return null;
};

DailyEvents.propTypes = {
  displayHours: isValidHour,
  startHour: isValidHour
};

DailyEvents.defaultProps = {
  displayHours: 24,
  startHour: 7
};

export default DailyEvents;
