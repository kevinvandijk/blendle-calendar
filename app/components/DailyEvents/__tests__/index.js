import React from 'react';
import renderer from 'react-test-renderer';
import DailyEvents from '../';

describe('Components', () => {
  describe('DailyEvents', () => {
    it('Renders with defaults', () => {
      const tree = renderer.create(
        <DailyEvents />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('Changes the amount of displayed hours and the starting hour', () => {
      const tree = renderer.create(
        <DailyEvents displayHours={ 12 } startHour={ 9 } />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('Renders a scrollable list', () => {
      const tree = renderer.create(
        <DailyEvents scrollable />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('Renders given appointments and looks at the time only (only care about start and end hour)', () => {
      const appointments = [
        {
          id: 1,
          start: '2017-02-08 08:00',
          end: '2017-02-08 17:00',
          title: 'Being awake'
        },
        {
          id: 2,
          start: '2017-02-08 07:00',
          end: '2017-02-08 09:20',
          title: 'Wake up'
        },
        {
          id: 3,
          start: '2017-02-08 07:00',
          end: '2017-02-08 08:43',
          title: 'Snooze'
        },
        {
          id: 4,
          start: '2017-02-08 09:20',
          end: '2017-02-08 10:00',
          title: 'Throw out alarm clock'
        }
      ];

      const tree = renderer.create(
        <DailyEvents appointments={ appointments } />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
