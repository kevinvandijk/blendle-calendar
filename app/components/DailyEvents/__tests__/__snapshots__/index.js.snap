exports[`Components DailyEvents Changes the amount of displayed hours and the starting hour 1`] = `
<div
  className="DailyEvents">
  <div
    className="DailyEvents-list">
    <li
      className="DailyEvents-Hour">
      <time>
        09:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        10:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        11:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        12:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        13:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        14:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        15:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        16:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        17:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        18:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        19:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        20:00
      </time>
    </li>
  </div>
</div>
`;

exports[`Components DailyEvents Renders a scrollable list 1`] = `
<div
  className="DailyEvents">
  <div
    className="DailyEvents-list DailyEvents-list--is-scrollable">
    <li
      className="DailyEvents-Hour">
      <time>
        07:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        08:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        09:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        10:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        11:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        12:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        13:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        14:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        15:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        16:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        17:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        18:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        19:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        20:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        21:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        22:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        23:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        00:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        01:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        02:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        03:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        04:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        05:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        06:00
      </time>
    </li>
  </div>
</div>
`;

exports[`Components DailyEvents Renders with defaults 1`] = `
<div
  className="DailyEvents">
  <div
    className="DailyEvents-list">
    <li
      className="DailyEvents-Hour">
      <time>
        07:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        08:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        09:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        10:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        11:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        12:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        13:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        14:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        15:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        16:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        17:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        18:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        19:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        20:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        21:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        22:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        23:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        00:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        01:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        02:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        03:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        04:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        05:00
      </time>
    </li>
    <li
      className="DailyEvents-Hour">
      <time>
        06:00
      </time>
    </li>
  </div>
</div>
`;
